scenarioo-client/app/manage/branch-aliases/branch-aliases.component.ts

```diff
@@ -0,0 +1,89 @@
+import {Component, OnInit} from '@angular/core';
+import {BranchAliasesResource} from '../../shared/services/branchAliasResource.service';
+import {BranchesResource} from '../../shared/services/branchesResource.service';
+import {IBranchAlias, IBranchBuilds} from '../../generated-types/backend-types';
+import {map} from 'rxjs/operators';
+import {Observable} from 'rxjs';
+
+@Component({
+    selector: 'sc-branch-aliases',
+    template: require('./branch-aliases.component.html'),
+    styles: [require('./branch-aliases.component.css').toString()],
+})
+export class BranchAliasesComponent implements OnInit {
+
+    branchAliases: IBranchAlias[] = [];
+    branches$: Observable<IBranchBuilds[]>;
+    successfullyUpdatedBranchAliases = false;
```

Comment:
How far do we go with declaring the types of variables/return types of methods in TypeScript? Is there some kind of convention e. g. "everything not primitive"?

Reply:
I think the linter does it as you suggested. I think it's also valid to specifiy primitive types. Some ppl says if you directly assign it, no type is requiered.